File: india_compliance/gst_india/report/gst_itemised_sales_register/gst_itemised_sales_register.py
# Copyright (c) 2013, DontManage and contributors
# For license information, please see license.txt

from dontmanage import _
from dontmanageerp.accounts.report.item_wise_sales_register.item_wise_sales_register import (
    _execute,
)

from india_compliance.gst_india.report.gst_sales_register.gst_sales_register import (
    get_additional_table_columns,
    get_column_names,
)


def execute(filters=None):
    additional_table_columns = get_additional_table_columns()
    additional_table_columns.append(
        {
            "fieldtype": "Data",
            "label": _("HSN Code"),
            "fieldname": "gst_hsn_code",
            "width": 120,
        }
    )

    return _execute(
        filters,
        additional_table_columns,
        get_column_names(additional_table_columns),
    )

File: india_compliance/gst_india/report/gst_itemised_sales_register/gst_itemised_sales_register.js
// Copyright (c) 2016, DontManage and contributors
// For license information, please see license.txt
/* eslint-disable */

{% include "dontmanageerp/accounts/report/item_wise_sales_register/item_wise_sales_register.js" %}
{% include "india_compliance/gst_india/report/utils.js" %}

let filters = dontmanage.query_reports["Item-wise Sales Register"]["filters"];

// Add GSTIN filter
filters = filters.concat({
    "fieldname":"company_gstin",
    "label": __("Company GSTIN"),
    "fieldtype": "Select",
    "placeholder":"Company GSTIN",
    "options": [""],
    "width": "80"
}, {
    "fieldname":"invoice_type",
    "label": __("Invoice Type"),
    "fieldtype": "Select",
    "placeholder":"Invoice Type",
    "options": ["", "Regular", "SEZ", "Export", "Deemed Export"]
});

// Handle company on change
for (var i = 0; i < filters.length; ++i) {
    if (filters[i].fieldname === 'company') {
        filters[i].on_change = fetch_gstins;
    }
}

dontmanage.query_reports["GST Itemised Sales Register"] = { "filters": filters, "onload": fetch_gstins };
